Anagram/Anagram.cs

```diff
@@ -0,0 +1,86 @@
+            return this.wordList.Length;
+        }
+        public int FillMatchedWords(int totalWords)
+        {
+            String inputString = inputWordBox.Text;
+            int inputStringLength = inputString.Length;
+
+            char[] inputStringChars = inputString.ToUpper().ToCharArray();
+            Array.Sort(inputStringChars);
+
+            String sortedInputString = new string(inputStringChars);
+            for (int m = 0; m < totalWords; m++)
+            {
+                String comparedWord = this.wordList[m];
+
+                if (inputStringLength == comparedWord.Length)
+                {
+                    char[] comparedWordChars = comparedWord.ToUpper().ToCharArray();
+                    Array.Sort(comparedWordChars);
+                    String sortedComparedWord = new String(comparedWordChars);
+                    if (sortedInputString != sortedComparedWord)
+                        continue;
+                    matchedWordsListFinal.Add(comparedWord);
+
+                }
+                else
+                {
+                    for (int n = 0; n < totalWords; n++)
+                    {
+                        int nextWordOfListLength = this.wordList[n].Length;
+                        int concatenatedWordsLength = comparedWord.Length + nextWordOfListLength;
+
+                        if (nextWordOfListLength == 0 || comparedWord.Length == 0)
+                            continue;
+                        if (inputStringLength != concatenatedWordsLength)
+                            continue;
+                        char[] comparedWordChar = comparedWord.ToUpper().ToCharArray();
+                        char[] nextWordOfList = this.wordList[n].ToUpper().ToCharArray();
+                        List<char> concatenatedLengths = new List<char>(comparedWordChar.Length + nextWordOfList.Length);
+                        concatenatedLengths.AddRange(comparedWordChar);
+                        concatenatedLengths.AddRange(nextWordOfList);
+                        concatenatedLengths.ToArray();
+                        char[] concatenatedChars = concatenatedLengths.ToArray();
+                        Array.Sort(concatenatedChars);
+                        String concatenatedComparedWord = new string(concatenatedChars);
+
+                        if (sortedInputString != concatenatedComparedWord)
+                            continue;
+                        matchedWordsListFinal.Add(comparedWord + " + " + this.wordList[n]);
+                    }
+                }
+            }
+            return this.matchedWordsListFinal.Count;
+        }
+        private void Anagrams_Click(object sender, EventArgs e)
+        {
+            if (inputWordBox.Text == "")
+            {
+                MessageBox.Show("First, enter a word to be checked for anagrams!");
+            }
+
+            int totalWords = WordLoad();
```

Comment:
both methods are public which means i can call the fill before load. Is that expected? this creates some temporal dependency between the methods...